src/signalSTM.h

```diff
@@ -0,0 +1,354 @@
+SimpleFIFO<int,FIFO_LENGTH> FiFo; //store FIFO_LENGTH # ints
+SignalDetectorClass musterDec;
+
+
+#include <EEPROM.h>
+#include "cc1101.h"
+
+volatile bool blinkLED = false;
+//String cmdstring = "";
+volatile unsigned long lastTime = micros();
+bool hasCC1101 = false;
+char IB_1[14]; // Input Buffer one - capture commands
+
+
+
+
+
+void setup() {
+
+	pinAsOutput(PIN_WIZ_RST);
+
+	Serial.begin(BAUDRATE);
+	while (!Serial) {
+		; // wait for serial port to connect. Needed for native USB
+	}
+
+	#ifdef MAPLE_WATCHDOG
+		if (IWatchdog.isReset(true)) {
+			MSG_PRINTLN(F("Watchdog caused a reset"));
+			watchRes = true;
+		}
+		else {
+			watchRes = false;
+		}
+		IWatchdog.begin(20000000);	// Init the watchdog timer with 20 seconds timeout
+		if (IWatchdog.isEnabled()) {
+			MSG_PRINTLN(F("Watchdog enabled"));
+		}
+#endif
+
+	//delay(2000);
+	pinAsInput(PIN_RECEIVE);
+	pinAsOutput(PIN_LED);
+	// CC1101
+
+	//wdt_reset();
+
+	#ifdef CMP_CC1101
+		cc1101::setup();
+	#endif
+	initEEPROM();
+	#ifdef CMP_CC1101
+		DBG_PRINT(FPSTR(TXT_CCINIT));
+		// for variant Circuit board for 4 cc110x and compatible version with 1 cc110x
+		#if defined(DEBUG)
+			DBG_PRINT(FPSTR(("(misoPin="))); DBG_PRINT((misoPin));
+			DBG_PRINT(FPSTR((" mosiPin="))); DBG_PRINT((mosiPin));
+			DBG_PRINT(FPSTR((" sckPin="))); DBG_PRINT((sckPin));
+			DBG_PRINT(FPSTR((" csPin="))); DBG_PRINT((csPin));
+			DBG_PRINTLN(")");
+		#endif
+
+		cc1101::CCinit();                   // CC1101 init
+		hasCC1101 = cc1101::checkCC1101();  // Check for cc1101
+
+
+		if (hasCC1101)
+		{
+			//DBG_PRINTLN("CC1101 found");
+			DBG_PRINT(FPSTR(TXT_CC1101)); DBG_PRINTLN(FPSTR(TXT_FOUND));
+			musterDec.setRSSICallback(&cc1101::getRSSI);          // Provide the RSSI Callback
+		} else {
+			musterDec.setRSSICallback(&rssiCallback);             // Provide the RSSI Callback
+		}
+	#endif
+
+	pinAsOutput(PIN_SEND);
+	DBG_PRINTLN(F("Starting timerjob"));
+	delay(50);
+
+#if defined(ARDUINO) && ARDUINO <= 100                            // to compile with PlatformIO
+	TIM_TypeDef *Instance = TIM1;
+	HardwareTimer *MyTim = new HardwareTimer(Instance);
+	MyTim->setMode(2, TIMER_OUTPUT_COMPARE);
+	MyTim->setOverflow(31*1000, MICROSEC_FORMAT);
+	MyTim->attachInterrupt(cronjob);
+	MyTim->resume();
+#endif
+
+	/*MSG_PRINT("MS:"); 	MSG_PRINTLN(musterDec.MSenabled);
+	MSG_PRINT("MU:"); 	MSG_PRINTLN(musterDec.MUenabled);
+	MSG_PRINT("MC:"); 	MSG_PRINTLN(musterDec.MCenabled);*/
+	//cmdstring.reserve(40);
+
+	musterDec.setStreamCallback(&writeCallback);
+
+
+#ifdef CMP_CC1101
+	if (!hasCC1101 || cc1101::regCheck()) {
+#endif
+	enableReceive();
+	DBG_PRINT(FPSTR(TXT_RECENA));
+#ifdef CMP_CC1101
+	}
+	else {
+		DBG_PRINT(FPSTR(TXT_CC1101));
+		DBG_PRINT(FPSTR(TXT_DOFRESET));
+		DBG_PRINTLN(FPSTR(TXT_COMMAND));
+	}
+#endif
+	MSG_PRINTER.setTimeout(400);
+}
+
+
+
+#if ARDUINO < 190                   /* MR neeed ??? - ToDo more information */
+	void cronjob(HardwareTimer*) {
+#else
+	void cronjob() {
```

Comment:
Die Ausführung der cronjob Funktion entspricht nicht der Variante vom ESP oder Arduino. Ist das absicht oder basiert die einfach auf einem veralteten Fork?

PS: Die Einrückung stimmt bei mir nicht, ist das bei dir auch so?

Reply:
Die Einrückung sieht normal aus in beiden Oberflächen.

Funktion cronjob Absicht oder alter fork?
Da kann ich nur wieder zugeben, dies habe ich so übernommen.

Reply:
Ich könnte jetzt hier an deinem Code Anpassung vornehmen, halte mich aber erst mal zurück.

Ich hab den cronjob mal angepasst, so dass er nicht die vollen 31 ms Timeout wartet, wenn schon eine gewisse Zeit nach dem letzten Interrupt vergangen ist.

Reply:
Du kannst gern Anpassungen vornehmen oder poste sie hier.
Ich kämpfe soeben das der Code in Arduino IDE + Platform IO anerkannt wird.

Reply:
Verwenden wir denn für beide Varianten, Arduino IDE und plattformIO das gleiche Framework?
Wenn ja, dann müssen doch auch die gleichen Funktionen zur Verfügung stehen.

Fehlt vielleicht ein
#include "HardwareTimer.h"

Reply:
Verwenden wir denn für beide Varianten, Arduino IDE und plattformIO das gleiche Framework?

In Arduino IDE wüßte ich nicht wo ich das Framework entnehmen könnte?
Da geht es zu compilieren. In der PlatformIO steht `framework = arduino` drin aber er stößt sich an der Zeile
`MyTim->attachInterrupt(cronjob);`
Meldung: 
`src\signalSTM.h:166:25: error: invalid conversion from 'void (*)()' to 'void (*)(HardwareTimer*)' [-fpermissive]`

#include "HardwareTimer.h" brachte auch kein Erfolg.

Reply:
In der Plattformio Doku steht, dass es zwei voneinander unabhängige Frameworks gibt:
https://docs.platformio.org/en/latest/platforms/ststm32.html#configuration

Ich weiss ja nicht, welche Du in der Arduino IDE hinterlegt hast, aber für PlattformIO kommt STM32Duino  als Default zum Einsatz.

Es gibt dann noch die STM32 maple Version. Vermutlich ist das diese hier:
https://github.com/rogerclarkmelbourne/Arduino_STM32/wiki

Vertmutlich ist der Code nur für eines der beiden Frameworks geschrieben. Nur für welches?

Reply:
Es ist der
`board_build.core = STM32Duino                                  ; https://docs.platformio.org/en/latest/platforms/ststm32.html#configuration`

Der Framework andere von rogerclarkbourne wird es nicht sein. Dieser bringt noch mehr Fehler und ich hatte auch schonmal über Arduino IDE verifiziert, das es der STM32Duino ist. Die Arduino IDE verwendet auch die Bibliothek vom Orginal.

Entweder es fehlt noch eine Angabe speziell oder ich übersehe was.

Aufruf:
`MyTim->attachInterrupt(cronjob);`
Fehler:
```
main.cpp:(.text.setup+0x220): undefined reference to `cronjob()'
collect2.exe: error: ld returned 1 exit status
```

Reply:
hab gerade dein Repo HomeAutoUser:dev-r3.4_plattformIO_maple_mini ausgecheckt und alle dev_maple_mini Varianten erfolgreich compiliert.

Habe das Framework in Vesion 7 verwendet:
https://community.platformio.org/t/st-stm32-dev-platform-7-0-0-new-boards-improved-support-for-cmsis-framework-updated-arduino-core/14584?utm_source=platformio&utm_medium=piohome

Reply:
;-) Leider sind in dem Branch genau die Zeilen
`if ARDUINO ... noch drin.`
Im Branch dev-r3.4_plattformIO_maple_mini-revised siehst du, das es nicht geht.

Stolperfalle ist und bleibt die Zeile
`MyTim->attachInterrupt(cronjob);`

So Feier der Nachbarn zu Ende und ich kann nun schlafen gehen.

Reply:
Compiliert bei mir auch.

Reply:
ich passe gleich noch was an und mache einen commit